src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotyper.java

```diff
@@ -144,47 +153,68 @@
+            doc = "Input targets file. A tab-separated (.tsv) list of targets (not a BED file) with at least" +
+                    " the following header columns: contig, start, stop, name. It must be a subset or can be all of the" +
+                    " targets included in the read counts table.",
             fullName = ExomeStandardArgumentDefinitions.TARGET_FILE_LONG_NAME,
             shortName = ExomeStandardArgumentDefinitions.TARGET_FILE_SHORT_NAME,
             optional = true
     )
     protected File inputTargetListFile = null;
 
+    @ArgumentCollection
+    protected OptionalIntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
     @Override
     protected Object doWork() {
         /* check args */
         IOUtils.canReadFile(inputRawReadCountsFile);
         IOUtils.canReadFile(inputContigAnnotsFile);
-        Utils.validateArg(baselineMappingErrorProbability > 0 && baselineMappingErrorProbability < 1, "Must have 0 < baseline mapping error probability < 1.");
+        Utils.validateArg(baselineMappingErrorProbability > 0 && baselineMappingErrorProbability < 1,
+                "Must have 0 < baseline mapping error probability < 1.");
+
         /* read input read counts */
         final ReadCountCollection rawReadCounts;
         try {
             logger.info("Parsing raw read count collection file...");
             rawReadCounts = ReadCountCollectionUtils.parse(inputRawReadCountsFile);
         } catch (final IOException ex) {
-            throw new UserException.CouldNotReadInputFile("Could not read raw read count collection file");
+            throw new UserException.CouldNotReadInputFile("Could not read the raw read count collection file");
         }
 
         /* parse contig genotype ploidy annotations */
         final List<ContigGermlinePloidyAnnotation> contigGermlinePloidyAnnotationList;
         logger.info("Parsing contig genotype ploidy annotations file...");
         contigGermlinePloidyAnnotationList = ContigGermlinePloidyAnnotationTableReader.readContigGermlinePloidyAnnotationsFromFile(inputContigAnnotsFile);
 
-
         /* parse target list */
         final List<Target> inputTargetList;
         if (inputTargetListFile != null) {
             inputTargetList = TargetTableReader.readTargetFile(inputTargetListFile);
         } else {
-            logger.info("Target list was not provided -- getting target list from the read count collection.");
+            logger.info("No target list was provided; using all targets present in the read count collection");
             inputTargetList = rawReadCounts.targets();
         }
+        final List<SimpleInterval> genotypingIntervals;
+        if (intervalArgumentCollection.intervalsSpecified()) {
+            final SAMSequenceDictionary sequenceDictionary = generateMaximallyCoveringSAMSequenceDictionary(inputTargetList);
```

Comment:
This is essentially what happens when a dictionary is created from an index file. This method short-circuits the index file creation. I have carefully studied `OptionalIntervalArgumentCollection` and what we are doing here causes no issues.